src/main/scala/akka/persistence/cassandra/query/CassandraReadJournalProvider.scala

```diff
@@ -0,0 +1,24 @@
+package akka.persistence.cassandra.query
+
+import akka.actor.ExtendedActorSystem
+import akka.persistence.query.ReadJournalProvider
+import com.typesafe.config.Config
+import akka.persistence.query.javadsl.ReadJournal
+import scala.util.control.NonFatal
+
+class CassandraReadJournalProvider(system: ExtendedActorSystem, config: Config) extends ReadJournalProvider {
+
+  override val scaladslReadJournal: scaladsl.CassandraReadJournal =
+    try {
+      new scaladsl.CassandraReadJournal(system, config)
+    } catch {
+      case NonFatal(e) =>
+        // TODO can be removed when https://github.com/akka/akka/issues/18976 is fixed
+        system.log.error(e, "Failed to initialize CassandraReadJournal")
```

Comment:
Good idea. The exception is otherwise hidden which can be really annoying for the user.